src/sync/installation.ts

```diff
@@ -446,65 +446,64 @@ export async function maybeScheduleNextTask(
+	const inProgressStorage = new RedisInProgressStorageWithTimeout(redis);
+	const deduplicator = new Deduplicator(
+		inProgressStorage, 1_000
+	);
 
-				sentry.setUser({
-					gitHubInstallationId: installationId,
-					jiraHost
-				});
+	return async (data: BackfillMessagePayload, sentry: Hub, logger: Logger): Promise<void> => {
+		const { installationId, jiraHost } = data;
 
-				const nextTaskDelaysMs: Array<number> = [];
+		logger.child({ gitHubInstallationId: installationId, jiraHost });
 
-				const result = await deduplicator.executeWithDeduplication(
-					"i-" + installationId + "-" + jiraHost,
-					() => doProcessInstallation(app, data, sentry, installationId, jiraHost, logger, (delay: number) =>
-						nextTaskDelaysMs.push(delay)
-					));
+		try {
+			if (await isBlocked(installationId, logger)) {
+				logger.warn("blocking installation job");
+				return;
+			}
 
-				switch (result) {
-					case DeduplicatorResult.E_OK:
-						logger.info("Job was executed by deduplicator");
-						maybeScheduleNextTask(data, nextTaskDelaysMs, logger);
-						break;
-					case DeduplicatorResult.E_NOT_SURE_TRY_AGAIN_LATER: {
-						logger.warn("Possible duplicate job was detected, rescheduling");
-						await sqsQueues.backfill.sendMessage(data, RETRY_DELAY_BASE_SEC, logger);
-						break;
-					}
-					case DeduplicatorResult.E_OTHER_WORKER_DOING_THIS_JOB: {
-						logger.warn("Duplicate job was detected, rescheduling");
-						// There could be one case where we might be losing the message even if we are sure that another worker is doing the work:
-						// Worker A - doing a long-running task
-						// Redis/SQS - reports that the task execution takes too long and sends it to another worker
-						// Worker B - checks the status of the task and sees that the Worker A is actually doing work, drops the message
-						// Worker A dies (e.g. node is rotated).
-						// In this situation we have a staled job since no message is on the queue an noone is doing the processing.
-						//
-						// Always rescheduling should be OK given that only one worker is working on the task right now: even if we
-						// gather enough messages at the end of the queue, they all will be processed very quickly once the sync
-						// is finished.
-						await sqsQueues.backfill.sendMessage(data, RETRY_DELAY_BASE_SEC + RETRY_DELAY_BASE_SEC * Math.random(), logger);
-						break;
-					}
+			sentry.setUser({
+				gitHubInstallationId: installationId,
+				jiraHost
+			});
+
+			const nextTaskDelaysMs: Array<number> = [];
+
+			const result = await deduplicator.executeWithDeduplication(
+				"i-" + installationId + "-" + jiraHost,
+				() => doProcessInstallation(app, data, sentry, installationId, jiraHost, logger, (delay: number) =>
+					nextTaskDelaysMs.push(delay)
+				));
+
+			switch (result) {
+				case DeduplicatorResult.E_OK:
+					logger.info("Job was executed by deduplicator");
+					maybeScheduleNextTask(data, nextTaskDelaysMs, logger);
```

Comment:
Do you want to chuck an `await` here while your in the neighbourhood `maybeScheduleNextTask(data, nextTaskDelaysMs, logger);`

Reply:
actually, this is done on purpose.  We don't want to wait as it's just putting another message on the queue